app/usecases/reactions/update_materials.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+# Sample Structure
+class OSample < OpenStruct
+  def initialize(data)
+    # set nested attributes
+
+    %w[residues elemental_compositions].each do |prop|
+      prop_value = data.delete(prop) || []
+
+      prop_value.each { |i| i.delete :id }
+
+      data.merge!(
+        "#{prop}_attributes" => prop_value
+      ) unless prop_value.blank?
+    end
+
+    data['elemental_compositions_attributes'].each { |i| i.delete('description') } if data['elemental_compositions_attributes']
+    data['show_label'] = false if data['show_label'].blank?
+    super
+  end
+
+  def is_new
+    to_boolean super
+  end
+
+  def is_split
+    to_boolean super
+  end
+
+  def to_boolean(string)
+    !!"#{string}".match(/^(true|t|yes|y|1)$/i)
+  end
+end
+
+module Usecases
+  module Reactions
+    class UpdateMaterials
+      include ContainerHelpers
```

Comment:
@jhmegorei, using `include` makes it so that `ContainerHelplers` isn't treated as a Grape helper anymore like previously : https://github.com/ComPlat/chemotion_ELN/blob/e412ff7f8c61f2c347458cfc9b0ca6d2f2ffb99b/app/api/chemotion/reaction_api.rb#L220

Therefore `update_datamodel` looses access to the Grape namespace / scope, which causes `./spec/features/sync_collection_spec.rb:56` to break:

```
     NameError:
       undefined local variable or method `current_user' for #<Usecases::Reactions::UpdateMaterials:0x000055acd0812068>
     # ./app/api/helpers/container_helpers.rb:117:in `can_update_container'
     # ./app/api/helpers/container_helpers.rb:27:in `create_or_update_containers'
     # ./app/api/helpers/container_helpers.rb:18:in `update_datamodel'
     # ./app/usecases/reactions/update_materials.rb:160:in `update_existing_sample'
     # ./app/usecases/reactions/update_materials.rb:69:in `block (3 levels) in execute!'
     # ./app/usecases/reactions/update_materials.rb:59:in `each'
     # ./app/usecases/reactions/update_materials.rb:59:in `each_with_index'
     # ./app/usecases/reactions/update_materials.rb:59:in `block (2 levels) in execute!'
     # ./app/usecases/reactions/update_materials.rb:56:in `each'
     # ./app/usecases/reactions/update_materials.rb:56:in `block in execute!'
     # ./app/usecases/reactions/update_materials.rb:54:in `execute!'
     # ./app/api/chemotion/reaction_api.rb:173:in `block (3 levels) in <class:ReactionAPI>'
```

Ideas on how to fix this?

Reply:
ContainerHelpers is just a ruby module, so writing `include ContainerHelpers` should work (constant resolution issues aside). Thing is that ContainerHelpers extends Grape::Api::Helpers, which in turn does some stuff that might interfere with your object if not careful.

The error message was caused by `current_user` not being  available in your usecase class. This can easily be rectified by adding the current_user as an attr_reader to your usecase class and passing it in the initializer

```
class Usecases::Reactions::UpdateMaterials
  attr_reader :current_user

  def initialize(someStuff, current_user:)
    @current_user = current_user
  end
end
```

Reply:
> This can easily be rectified by adding the current_user as an attr_reader to your usecase class

Thanks for the pointer! All tests (Ruby unit & acceptance) pass now locally.